lib/ngx/balancer.md

```diff
@@ -237,6 +237,29 @@ This function was first added in the `0.1.7` version of this library.
 
 [Back to TOC](#table-of-contents)
 
+recreate_request
+------------
+**syntax:** `ok, err = balancer.recreate_request()`
+
+**context:** *balancer_by_lua&#42;*
+
+Regenerates the request buffer for sending to the upstream server. This is useful, for example
+if you want to change a request header field to the new upstream server on balancer retries.
+
+Normally this does not work because the request buffer is created once during upstream module
+initialization and won't be regenerated for subsequent retries. However you can use
+`proxy_set_header My-Header $my_header` and sets the `ngx.var.my_header` variable inside the
+balancer phase. Calling this function after that will cause the request buffer to be re-generated
```

Comment:
"Calling this function after that" can be confusing for the reader. "Calling `balancer.recreate_request()` after updating a header field will cause..."